src/datumaro/components/dataset_storage.py

```diff
@@ -420,8 +471,13 @@ def put(self, item: DatasetItem) -> None:
 
         if is_new and not self.is_cache_initialized():
             self._length = None
+            self._set_of_ann_types = set()
         if self._length is not None:
             self._length += is_new
+            # # when adding a new item, task_type will be updated automatically
+            # for ann in item.annotations:
+            #     self._set_of_ann_types.add(ann.type)
+            # self._task_type = TaskAnnotationMapping().get_task(self._set_of_ann_types)
```

Comment:
I think this can be deleted.

Reply:
fixed at 0a313ca5d4b9e08208b20a226840d026456c1655